sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_test.cpp

```diff
@@ -21,6 +22,36 @@ TEST(KeyClient, initClient)
   {
     KeyClientOptions options;
     options.Retry.MaxRetries = 10;
-    EXPECT_NO_THROW(KeyClient keyClient("vaultUrl", credential));
+    EXPECT_NO_THROW(KeyClient keyClient("vaultUrl", credential, options));
+  }
+}
+
+TEST(KeyClient, ServiceVersion)
+{
+  auto credential
+      = std::make_shared<Azure::Identity::ClientSecretCredential>("tenantID", "AppId", "SecretId");
+  {
+    // 7.0
+    EXPECT_NO_THROW(auto options = KeyClientOptions(ServiceVersion::V7_0);
+                    KeyClient keyClient("vaultUrl", credential, options);
+                    EXPECT_EQ(options.GetVersionString(), "7.0"););
+  }
+  {
+    // 7.1
+    EXPECT_NO_THROW(auto options = KeyClientOptions(ServiceVersion::V7_1);
+                    KeyClient keyClient("vaultUrl", credential, options);
+                    EXPECT_EQ(options.GetVersionString(), "7.1"););
+  }
+  {
+    // 7.2
+    EXPECT_NO_THROW(auto options = KeyClientOptions(ServiceVersion::V7_2);
+                    KeyClient keyClient("vaultUrl", credential, options);
+                    EXPECT_EQ(options.GetVersionString(), "7.2"););
+  }
+  {
+    // arbitrary version
+    EXPECT_NO_THROW(auto options = KeyClientOptions(ServiceVersion("1.0"));
+                    KeyClient keyClient("vaultUrl", credential, options);
+                    EXPECT_EQ(options.GetVersionString(), "1.0"););
```

Comment:
We allow this? What happens if I send a request to an incorrect (or unsupported) service version?

Reply:
We would get RequestFailedException from the service complaining about it.